tests/lib/ocpp/v16/test_composite_schedule.cpp

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2024 Pionix GmbH and Contributors to EVerest
+
+#include <filesystem>
+#include <fstream>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <nlohmann/json.hpp>
+#include <vector>
+namespace fs = std::filesystem;
+
+#include "everest/logging.hpp"
+#include <database_handler_mock.hpp>
+#include <evse_security_mock.hpp>
+#include <ocpp/common/call_types.hpp>
+#include <ocpp/common/evse_security_impl.hpp>
+#include <ocpp/v16/charge_point_impl.hpp>
+#include <ocpp/v16/enums.hpp>
+#include <ocpp/v16/smart_charging.hpp>
+#include <optional>
+
+namespace ocpp {
+namespace v16 {
+
+/**
+ * CompositeSchedule Test Fixture
+ */
+class CompositeScheduleTestFixture : public testing::Test {
+protected:
+    void SetUp() override {
+        this->evse_security = std::make_shared<EvseSecurityMock>();
+    }
+
+    void addConnector(int id) {
```

Comment:
The community prefers snake_case for function names. I am not sure that we are consistent with this in our test files thought. If not you can ignore this comment.